src/puya/ir/arc4_router.py

```diff
@@ -698,12 +701,7 @@ def create_abi_router(
         source_location=router_location,
         args=[],
         return_type=wtypes.bool_wtype,
-        body=create_block(
-            router_location,
-            "abi_bare_routing",
-            *router,
-            reject(contract.source_location),
-        ),
+        body=create_block(router_location, "abi_bare_routing", *router),
```

Comment:
I removed the 'fall through' `reject` here in favour of explicitly terminating the blocks created by `route_abi_methods`